src/shared/codelens/samTemplateCodeLensProvider.ts

```diff
@@ -6,44 +6,61 @@
 import * as _ from 'lodash'
 import * as vscode from 'vscode'
 import { TemplateFunctionResource, TemplateSymbolResolver } from '../cloudformation/templateSymbolResolver'
-import { getReferencedTemplateResources, LaunchConfiguration } from '../debug/launchConfiguration'
-import { TEMPLATE_TARGET_TYPE } from '../sam/debugger/awsSamDebugConfiguration'
+import { getConfigsMappedToTemplates, LaunchConfiguration } from '../debug/launchConfiguration'
+import { TEMPLATE_TARGET_TYPE, API_TARGET_TYPE } from '../sam/debugger/awsSamDebugConfiguration'
 import { AddSamDebugConfigurationInput } from '../sam/debugger/commands/addSamDebugConfiguration'
 import { localize } from '../utilities/vsCodeUtils'
 
 /**
- * Provides Code Lenses for generating debug configurations for SAM templates.
+ * Provides "Add Debug Configuration" CodeLenses to SAM template.yaml files,
+ * for resources that do not already have a mapped config in launch.json.
  */
 export class SamTemplateCodeLensProvider implements vscode.CodeLensProvider {
     public async provideCodeLenses(
         document: vscode.TextDocument,
         token: vscode.CancellationToken,
         symbolResolver = new TemplateSymbolResolver(document),
-        launchConfig = new LaunchConfiguration(document.uri)
+        launchConfig = new LaunchConfiguration(document.uri),
+        waitForSymbols: boolean = false
     ): Promise<vscode.CodeLens[]> {
-        const functionResources = await symbolResolver.getFunctionResources()
-        if (_(functionResources).isEmpty()) {
+        const apiResources = (await symbolResolver.getResourcesOfKind('api', waitForSymbols)) ?? []
```

Comment:
for this line and the one below, `getResourcesOfKind` can't return `undefined`?

Reply:
Probably just leftover. Fixed